src/pages/Dashboard/Profile.jsx

```diff
@@ -1,13 +1,294 @@
-import React from 'react';
+import React, { useCallback, useMemo, useReducer } from 'react';
+import styled from 'styled-components';
 import { AddressCard } from 'styled-icons/fa-regular/AddressCard';
 import { DashboardPage } from '../../components';
+import {
+    Button,
+    Card,
+    Checkbox,
+    Heading,
+    Input,
+    Select,
+} from '@cheapreats/react-ui';
+
+const testUser = {
+    status: 'PENDING',
+    email: 'jacky.chan@fake.utoronto.ca',
+    firstname: 'Jacky',
+    lastname: 'Chan',
+    gender: 'MALE',
+    school: 'UTSC',
+    bio: 'actor/fighter',
+    photo: 'https://i.ibb.co/HF6q3BF/doggo.gif',
+    links: [
+        {
+            name: 'github',
+            href: 'https://github.com/cheapreats/react-ui-library',
+        },
+        {
+            name: 'linkedin',
+            href: 'https://github.com/cheapreats/react-ui-library',
+        },
+        {
+            name: 'website',
+            href: 'https://github.com/cheapreats/react-ui-library',
+        },
+    ],
+};
+
+function reducer(state, action) {
+    switch (action.type) {
```

Comment:
Now with the changes handlers
You have a switch where by default u spread state and action, but for link do something special